plugins/kotlin/j2k/shared/tests/testData/newJ2k/postProcessing/GuardClause.java

```diff
@@ -1,8 +1,60 @@
 // IGNORE_K2
 public class Test {
-    void test(String s) {
-        if (s == null) {
+    void testRequire(String s1, boolean b1, boolean b2) {
+        if (s1 == null) {
             throw new IllegalArgumentException("s should not be null");
         }
+
+        if (!b1) {
+            throw new IllegalArgumentException();
+        } else {
+            System.out.println("never mind");
+        }
+        // comment above b2
+        if (b2) {
+            throw new IllegalArgumentException();
+        }
+
+        if (!b1 && b2) throw new IllegalArgumentException();
+        else {
+            System.out.println(1);
+            System.out.println(2);
+        }
+
+        if (s1.length() < 3) {
```

Comment:
Please make sure it works correctly with floating point. See

* https://youtrack.jetbrains.com/issue/KTIJ-13841

* `org.jetbrains.kotlin.idea.codeinsight.utils.NegatedBinaryExpressionSimplificationUtils#canBeSimplifiedWithoutChangingSemantics`